### PR TITLE
Engine: Centralize the place where value version is inferred.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
We move inference of value versions in a single place (when finalizing the `PartialTransacion`)
In an upcoming PR, we will infer the transaction version in the same place. 

This PR advances the state of #5164

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala [hunk 1]
```diff
@@ -28,7 +28,7 @@ final class Conversions(
   // We need the mapping for converting error message so we manually add it here.
   private val ptxCoidToNodeId = machine.ptx.nodes
     .collect {
-      case (nodeId, node: N.NodeCreate.WithTxValue[V.ContractId]) =>
+      case (nodeId, node @ N.NodeCreate(_, _, _, _, _, _)) =>
```
Comment (S11001001) [inline compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala:31]: (Follows up #206, #643.) Perhaps the definition of `Transaction.Value` should change, instead? Or,

```suggestion
      case (nodeId, node: N.NodeCreate[_, V.ContractId, _]) =>
```

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala [hunk 2]
```diff
@@ -471,46 +471,51 @@ final class Conversions(
...
 
     }
     builder.build
   }
 
-  def convertKeyWithMaintainers(
-      key: N.KeyWithMaintainers[V.VersionedValue[V.ContractId]],
+  def convertKeyWithMaintainers[Val](
+      convertValue: Val => Value,
+      key: N.KeyWithMaintainers[Val],
```
Comment (S11001001) [inline compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala:484]: More like `coYonedanvertKeyWithMaintainers`? :trollface:

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala [hunk 3]
```diff
@@ -580,6 +585,9 @@ final class Conversions(
 
   }
 
+  private def convertVersionnedValue(value: V.VersionedValue[V.ContractId]): Value =
```
Comment (S11001001) [inline compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala:588]:
```suggestion
  private def convertVersionedValue(value: V.VersionedValue[V.ContractId]): Value =
```

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala]: In the current state we should have to add a `WithPartialTransactionValue` type. 
I will see later if it is worthwhile to add it.

**Pull request merged**